MuonAnalysis/MomentumScaleCalibration/test/Macros/RooFit/MultiHistoOverlapAll_Base.C

```diff
@@ -333,13 +365,76 @@ void MultiHistoOverlapAll_Base_one(string files, string labels, string colors, s
     static_cast<float>(-TMath::Pi())
   };
 
+
+  // Plot 2D hystograms
+  // TO BE FIXED: something strange appears in the 2D plots for the sigma, this should be further investigated
+  Double_t zMass(91.1876);
+  Double_t Y1SMass(9.46);
+  Double_t maxDist(0.);
+  Double_t Mass=zMass;
+  Double_t fixrange=1.5;
+  if (resonance=="Y1S"){
+    Mass=Y1SMass;
+    fixrange=0.15;
+  }
+  for (int f=0; f<nfiles; f++){
+ 
+    if (file[f]!=0 && !file[f]->IsZombie()){
+
+      
+      TH2D* histo2D[2];
+      histo2D[0]= (TH2D*)file[f]->Get(histo2Dname[0]); //histoMassVsEtaPhiPlus
+      histo2D[1]= (TH2D*)file[f]->Get(histo2Dname[1]); //histoMassVsEtaPhiMinus
+      for(int s=0;s<2;s++){
+	TCanvas dummycanvas;
+	dummycanvas.SetFillColor(0);  
+	dummycanvas.cd()->SetTopMargin(0.07);
+	dummycanvas.cd()->SetRightMargin(0.15);
+	dummycanvas.cd()->SetLeftMargin(0.12);
+	histo2D[s]->SetTitle("");
+	histo2D[s]->GetZaxis()->SetLabelFont(42);
+	histo2D[s]->GetXaxis()->SetTitle(x2Dtitle[s]);
+	histo2D[s]->GetYaxis()->SetTitle(y2Dtitle[s]);
+	histo2D[s]->GetYaxis()->SetTitleOffset(0.9);
+	if(type=="mean"){//This is made to ensure that the range options are not used for the sigma
+	  Double_t zMin=Mass;
+	  Double_t zMax=Mass;
+ 
+	  for(int nbinX=1;nbinX<=histo2D[s]->GetNbinsX();nbinX++){
+	    for(int nbinY=1;nbinY<=histo2D[s]->GetNbinsY();nbinY++){
+	      Double_t value = histo2D[s]->GetBinContent(nbinX,nbinY);
+	      if(value<zMin)zMin=value;
+	      if(value>zMax)zMax=value;
+	    }
+	  }
+	  maxDist=fabs(zMax-Mass);
```

Comment:
@cardinia you are using here a variable `zMax` which is defined out of this scope (in the `if(type=="mean"){` clause)
How is this supposed to work?

Reply:
Dear Marco,

I am sorry but I do not understand what you mean: the zMax and zMin variables are defined within the scope of the condition regarding the type of the histogram, and the scope of that condition ends at line 415.

I defined the variables in that scope since they are necessary only to fix the z axis range when working with the average of the mass distribution and not the standard deviation.

Can you be more specific regarding the issue you noticed?

Thanks in advance,
Andrea

Reply:
Hi Marco,
no range is set for the sigma distribution, the plots are identical to what you would observe in the BiasCheck.root file drawing the histograms with the colz option.
I'm investigating the code in CompareBiasZValidation.cc (https://github.com/cms-sw/cmssw/blob/02d4198c0b6615287fd88e9a8ff650aea994412e/MuonAnalysis/MomentumScaleCalibration/test/Macros/RooFit/CompareBiasZValidation.cc#L16) to try and understand what causes the problem, as of now I did not find anything weird...
Anyway I put the comment you underlined in the code as a reminder that there is still work to be done, then I decided to make the pull request anyway given that some features can already be used in the framework.
Cheers,
Andrea